### PR TITLE
Update Spain holidays

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Update Spain holidays (checked with official sources for 2010-2023).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Review by arkid15r — COMMENTED** [review]
Nicely done! I believe the covered holiday dates precision will be drastically increased!

I noticed you're using (I guess intentionally) multiple identical ifs for different (chronologically within a year) holidays: `if self._year in {2013, 2019}:`, `if self._year in {2015, 2020}:`  Would it be possible to combine those or the current order is absolutely necessary for the resulting holiday list? If that's just a readability choice we should consider readabiity/performance tradeoffs before applying this approach widely.

Also some minor comments, mostly naming:

**Comment by KJhellico** [comment]
> I noticed you're using (I guess intentionally) multiple identical ifs for different (chronologically within a year) holidays: `if self._year in {2013, 2019}:`, `if self._year in {2015, 2020}:` Would it be possible to combine those or the current order is absolutely necessary for the resulting holiday list? If that's just a readability choice we should consider readabiity/performance tradeoffs before applying this approach widely.

No, order is not important, but in some of the following years one of these holidays may exist (in this province), and other may not. And `if` will have to be divided again. :)

File: holidays/countries/spain.py [hunk 1]
```diff
@@ -9,287 +9,657 @@
...
+            self._move_holiday(self._add_holiday_oct_12("Fiesta Nacional de España"))
+
+        if self._year in {2015, 2020}:
+            self._move_holiday(self._add_all_saints_day("Todos los Santos"))
+
+        if self._year in {2015, 2020}:
+            self._move_holiday(self._add_holiday_dec_6("Día de la Constitución Española"))
+
+        if self._year in {2013, 2019}:
+            self._move_holiday(self._add_immaculate_conception_day("Inmaculada Concepción"))
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply holidays/countries/spain.py:132]: I still prefer to keep chronological order of holidays throughout the year - for easier orientation.

[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/countries/spain.py:131]: Right, I know, I find it easier too. I just want to make sure the balance of compatibility/readability/performance we currently have is still there.

**Review by arkid15r — APPROVED** [review]

**Pull request merged**